go/coinstacks/osmosis/api/handlers.go

```diff
@@ -174,6 +191,150 @@ func (h *Handler) GetTxHistory(pubkey string, cursor string, pageSize int) (api.
 	return txHistory, nil
 }
 
-func (h *Handler) SendTx(hex string) (string, error) {
-	return h.httpClient.BroadcastTx(hex)
+func (h *Handler) SendTx(rawTx string) (string, error) {
+	return h.httpClient.BroadcastTx(rawTx)
+}
+
+func (h *Handler) EstimateGas(rawTx string) (string, error) {
+	return h.httpClient.GetEstimateGas(rawTx)
+}
+
+func (h *Handler) GetValidators() ([]cosmos.Validator, error) {
+	info, err := h.GetInfo()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get info")
+	}
+
+	apr, _, err := new(big.Float).Parse(info.(Info).APR, 10)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to parse apr: %s", apr)
+	}
+
+	return h.httpClient.GetValidators(apr)
+}
+
+func (h *Handler) GetValidator(address string) (*cosmos.Validator, error) {
+	info, err := h.GetInfo()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get info")
+	}
+
+	apr, _, err := new(big.Float).Parse(info.(Info).APR, 10)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to parse apr: %s", apr)
+	}
+
+	return h.httpClient.GetValidator(address, apr)
+}
+
+func (h *Handler) getAPRData() (*APRData, error) {
+	aprData := &APRData{}
+
+	g := new(errgroup.Group)
+
+	g.Go(func() error {
+		totalSupply, err := h.httpClient.GetTotalSupply("uosmo")
+		if err != nil {
+			return err
+		}
+
+		bTotalSupply, _, err := new(big.Float).Parse(totalSupply, 10)
+
+		aprData.totalSupply = totalSupply
+		aprData.bTotalSupply = bTotalSupply
+
+		return err
+	})
+
+	g.Go(func() error {
+		annualProvisions, err := h.httpClient.GetAnnualProvisions()
+		if err != nil {
+			return err
+		}
+
+		bAnnualProvisions, _, err := new(big.Float).Parse(annualProvisions, 10)
+
+		aprData.annualProvisions = annualProvisions
+		aprData.bAnnualProvisions = bAnnualProvisions
+
+		return err
+	})
+
+	g.Go(func() error {
+		communityTax, err := h.httpClient.GetCommunityTax()
+		if err != nil {
+			return err
+		}
+
+		bCommunityTax, _, err := new(big.Float).Parse(communityTax, 10)
+
+		aprData.communityTax = communityTax
+		aprData.bCommunityTax = bCommunityTax
+
+		return err
+	})
+
+	g.Go(func() error {
+		bondedTokens, err := h.httpClient.GetBondedTokens()
+		if err != nil {
+			return err
+		}
+
+		bBondedTokens, _, err := new(big.Float).Parse(bondedTokens, 10)
+
+		aprData.bondedTokens = bondedTokens
+		aprData.bBondedTokens = bBondedTokens
+
+		return err
+	})
+
+	err := g.Wait()
+
+	return aprData, err
+}
+
+func (h *Handler) getAccountData(pubkey string, denom string, apr *big.Float) (*AccountData, error) {
```

Comment:
Can you confirm that the account data is returned correctly as it stands?

Reply:
works in swagger docs. seems to be right at first glance:

```

  "balance": "16628631",
  "unconfirmedBalance": "0",
  "pubkey": "osmo1fx4jwv3aalxqwmrpymn34l582lnehr3eg40jnt",
  "accountNumber": 120668,
  "sequence": 112,
  "assets": [
    {
      "amount": "10329755410319112270",
      "denom": "gamm/pool/1"
    },
    {
      "amount": "723781",
      "denom": "ibc/27394FB092D2ECCD56123C74F36E4C1F926001CEADA9CA97EA622B25F41E5EB2"
    },
    {
      "amount": "122009731",
      "denom": "ibc/9712DBB13B9631EDFA9BF61B55F1B2D290B2ADB67E3A4EB3A875F3B6081B3B84"
    },
    {
      "amount": "1191860",
      "denom": "ibc/BE1BB42D4BE3C30D50B68D7C41DB4DFCE9678E8EF8C539F6E6A9345048894FCC"
    }
  ],
  "delegations": [
    {
      "validator": {
        "address": "osmovaloper1pjmngrwcsatsuyy8m3qrunaun67sr9x74vvvdk",
        "moniker": "Cypher Core",
        "jailed": false,
        
        ```